core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -454,6 +454,30 @@ public Table emptyCopy(int rowSize) {
         tables[1] = where(table2Selection);
         return tables;
     }
+    
+    /**
+     * Splits the table into two stratified samples, this uses the specified column to divide the table into groups, randomly assigning records to each according to the proportion given in trainingProportion. 
+     * 
+     *
+     * @param column the column to be used for the stratified sampling
+     * @param table1Proportion The proportion to go in the first table
+     * @return An array two tables, with the first table having the proportion specified in the method parameter,
+     * and the second table having the balance of the rows
+     */
+    public Table[] stratifiedSampleSplit(CategoricalColumn column, double table1Proportion){
+        Preconditions.checkArgument( CategoricalColumn.class.isInstance(this.column(column.name())),
```

Comment:
no need to check if it's a `CategoricalColumn`. we know it is from the method signature. I would do `containsColumn(column)` instead

Reply:
A user doing fancy stuff like table.nCol("xxx").asStringColumn() will cause issue

Reply:
If you do `.asStringColumn()` it will create a new column not in the table, so `containsColumn(column)` should still work in that case